pennylane/qchem/vibrational/localize_modes.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains functions to localize normal modes."""
+
+import warnings
+
+import numpy as np
+import scipy
+
+AU_TO_CM = 219475
+
+# pylint: disable=dangerous-default-value, too-many-statements
+
+
+def _pm_cost(q):
+    r"""Pipek-Mezey cost function whose minimization yields localized displacements."""
```

Comment:
Please put a reference such is wiki page or a paper.

Reply:
We need Args and Returns for private functions too. This applies to all other private functions as well.